hazelcast/src/main/java/com/hazelcast/topic/impl/TopicProxy.java

```diff
@@ -36,9 +41,11 @@
 
     protected static final String NULL_MESSAGE_IS_NOT_ALLOWED = "Null message is not allowed!";
     protected static final String NULL_LISTENER_IS_NOT_ALLOWED = "Null listener is not allowed!";
+    private final int partitionId;
 
     public TopicProxy(String name, NodeEngine nodeEngine, TopicService service) {
         super(name, nodeEngine, service);
+        this.partitionId = nodeEngine.getPartitionService().getPartitionId(getNameAsPartitionAwareData());
```

Comment:
partition id is used by `TotalOrderedTopicProxy` to make sure the total order. 
This one should not need it. 
See https://docs.hazelcast.org/docs/4.0.1/manual/html-single/index.html#understanding-topic-behavior